React/Modules/RCTUIManager.m

```diff
@@ -224,6 +224,8 @@ @implementation RCTUIManager
   NSDictionary *_componentDataByName;
 
   NSMutableSet<id<RCTComponent>> *_bridgeTransactionListeners;
+  NSMutableSet<id<RCTUIManagerObserver>> *_uiManagerObservers;
```

Comment:
I assume all native modules get deallocated at the same time, I guess the bridge is where it could leak.